### ISSUE TITLE
No way to tell when SCTP association is closed

### ISSUE THREAD

**Issue by taylor-b** [w3c/webrtc-pc#1612]
This was brought up in https://github.com/w3c/ortc/issues/755, for a call forking use case. You can see the discussion there for more details, but the short explanation is: there are situations where the SCTP association may be closed and the application needs to generate a new offer/answer to restore it. And there's currently no way to tell when this happens. In ORTC, RTCSctpTransport has a state enum and event, so I'd propose using that. The states are very straightforward so I don't anticipate it being an issue for implementers.

**Comment by taylor-b** [comment]
JSEP actually does call out this case of the answerer changing the SCTP port:

```
         If an SCTP association exists, and the remote SCTP port has
         changed, discard the existing SCTP association.  This includes
         the case when the PeerConnection state is "have-remote-
         pranswer".
```

So maybe we should reconsider marking this as CR-blocking.

The issue occurs when the SCTP port and TLS ID are changing due to a call being transitioned to a new remote endpoint (which is why "pranswer" would be used). But there's a chance the new endpoint uses the same SCTP port number as the previous endpoint. In which case the WebRTC endpoint would just see its SCTP association close, and would need to do a re-offer to establish the new association.

**Comment by stefhak** [comment]
@taylor-b perhaps you can provide a little more context/summary? (I tried to read the ortc conversation, but it was hard to follow)

At the face of it, this looks like something that could be handled between the ```neogtiationneeded``` event, the ```RTCDataChannelState``` and the different events on ```RTCDataChannel``` (```error```, ```close```).

**Comment by taylor-b** [comment]
I tried to summarize it in my previous comment; what part specifically needs more clarity?

Also, my reason for preferring an `RTCSctpTransport` state change as opposed to `negotiationneeded`:

> I'd prefer [an `RTCSctpTransport` state change event], because there also may be corner cases where the SCTP association is closed for some other reason [(besides this call forking scenario)], and renegotiating won't help. So better to let the application decide, given what it knows about the call topology. Plus, the state change event is already there in ORTC. 

And the `RTCDataChannel` events don't completely work, because they don't distinguish between an individual SCTP stream being closed and the whole association being closed.

**Comment by aboba** [comment]
@taylor-b Addition of an `RTCSctpTransport` state attribute and change event makes sense to me.
